### DISCUSSION TITLE
Lag / delay in reading out BK5000 images in Slicer

### DISCUSSION THREAD

**Discussion by LisanneVenix** [PlusToolkit/PlusLib#1094] [General]
We want to read out US images from our BK5000 device. With our EM-tracked US transducer, we want to do a landmark based registration with preoperative imaging. However, when reading out these images in slicer, there is a delay of about 1 to 3 seconds. This delay is not constant but varies. Below a list of the things we tried in order to fix this or to understand where the delay is coming from:

- We have another BK5000 device with other firmware, which does not have this delay. 
- When we use the settings continuousstreamingdisabled 'false' the delay decreases a lot, and is acceptable. 
- When we use OEM test tool with the command: QUERY:GRAB_FRAME "ON",30;, we do not have any delay.
- We also tried offline testing in plusserverlauncher, which seemed to work without a drop in framerate.

Therefore, our conclusion was that there is an issue with the connection between Plus and the BK device. Does anyone have an idea how to solve this?

Attached you can find the TRACE log file.

[060623_151137_PlusLog.txt](https://github.com/PlusToolkit/PlusLib/files/11665055/060623_151137_PlusLog.txt)

**Comment by lassoan** [comment]
For real-time image acquisition you need to have the streaming license from BK and use `QUERY:GRAB_FRAME "ON"` by enabling `ContinuousStreamingEnabled` option in Plus.

From the log file it seems that you are using an older Plus version. Does everything work well if you use the Plus "latest development snapshot" release?
